amp/src/test/java/org/digijava/kernel/ampapi/endpoints/activity/InterchangeUtilsTest.java

```diff
@@ -189,17 +191,32 @@ public void testFormatDateNullInput() throws Exception {
     }
 
     @Test
-    public void testParseDate() throws Exception {
-        assertEquals(new Date(124124124124L), InterchangeUtils.parseISO8601Date("1973-12-07T17:55:24.124+0300"));
+    public void testParseTimestamp() {
+        assertEquals(new Date(124124124124L), InterchangeUtils.parseISO8601DateTimestamp("1973-12-07T17:55:24.124+0300", true));
+    }
+    
+    @Test
+    public void testParseDate() {
+        assertEquals(new Date(124059600000L), InterchangeUtils.parseISO8601DateTimestamp("1973-12-07", false));
```

Comment:
Similar to having a dedicated date and timestamp formatter, it would be nice to have a dedicated parse for each of them, to avoid understanding why you need to send true/false ti the for `parseISO8601DateTimestamp`

Reply:
Agree with you